app/javascript/containers/PortfolioListContainer.jsx

```diff
@@ -1,37 +1,84 @@
-import React from 'react';
+import React from "react";
+import ReactModal from "react-modal";
 
-import { loadInitialState } from '../actions/initialState';
-import { bindActionCreators } from 'redux';
-import { connect } from 'react-redux';
-import { Link } from 'react-router-dom';
+import { loadInitialState } from "../actions/initialState";
+import { bindActionCreators } from "redux";
+import { connect } from "react-redux";
+import { Link } from "react-router-dom";
+import { post } from "../fetch/requester";
+import { addBuildingType } from "../actions/building_type";
 
 class PortfolioListContainer extends React.Component {
-  componentDidMount() {
-    if (window.INITIAL_STATE) {
-      this.props.initActions.loadInitialState(window.INITIAL_STATE);
+  constructor(props) {
+    super(props);
+    this.state = {
+      showModal: false
+    };
+    this.toggleModal = this.toggleModal.bind(this);
+    this.createBuildingType = this.createBuildingType.bind(this);
+  }
+
+  toggleModal() {
+    this.setState({ showModal: !this.state.showModal });
+  }
+
+  async createBuildingType(event) {
+    event.preventDefault();
+    const typeName = event.target.name.value;
+    try {
+      let response = await post("/api/building_types", {
+        name: typeName,
+        categories: []
+      });
+      const buildingType = { ...response.data, questions: [] };
+      const buildingTypeId = buildingType.id;
+      this.props.addBuildingType(buildingType);
+      this.props.history.push(`/building_types/${buildingTypeId}`);
+    } catch (error) {
+      console.log(error);
     }
   }
 
   render() {
     const portfolios = this.props.portfolios;
     const building_types = this.props.building_types;
-    return (<div>
-      <h2>Building Types</h2>
-      {Object.keys(building_types).map(id => {
-        return (
-          <p key={id}>
-            {building_types[id].name}
-            <Link to={`/building_types/${id}`}>Edit</Link>
-          </p>
-        )
-      })}
-      <h2>Portfolios</h2>
-      {Object.keys(portfolios).map(id => {
-        return (<p key={id}>{portfolios[id].name} |
-          <Link to={`/portfolios/${id}`}>Details</Link>
-        </p>)
-      })}
-    </div>);
+    return (
+      <div>
+        <h2>Building Types</h2>
+        <input
+          type="button"
+          value="Create New Building Type"
+          onClick={this.toggleModal}
+        />
+        <ReactModal isOpen={this.state.showModal}>
+          <form onSubmit={this.createBuildingType}>
+            <label>
+              Add Building Type
+              <input type="text" name="name" />
+            </label>
+            <input type="submit" value="Submit" />
```

Comment:
I noticed it's possible to create a building type with an empty string as the name, and also to have duplicate building type names, which can get confusing. Can we add some quick Rails validations to `building_type.rb` for uniqueness and presence of `name`? Make sure the error text shows in the modal as well (see previous comment) so the user knows what's wrong!